src/main/java/at/ac/tuwien/kr/alpha/common/program/impl/AnalyzedProgram.java

```diff
@@ -0,0 +1,44 @@
+package at.ac.tuwien.kr.alpha.common.program.impl;
+
+import org.apache.commons.lang3.tuple.ImmutablePair;
+
+import java.util.List;
+
+import at.ac.tuwien.kr.alpha.common.atoms.Atom;
+import at.ac.tuwien.kr.alpha.common.depgraph.ComponentGraph;
+import at.ac.tuwien.kr.alpha.common.depgraph.DependencyGraph;
+import at.ac.tuwien.kr.alpha.common.depgraph.SccResult;
+import at.ac.tuwien.kr.alpha.common.depgraph.StronglyConnectedComponentsHelper;
+import at.ac.tuwien.kr.alpha.common.rule.impl.InternalRule;
+
+public class AnalyzedProgram extends InternalProgram {
+
+	private final DependencyGraph dependencyGraph;
+	private final ComponentGraph componentGraph;
+
+	public AnalyzedProgram(List<InternalRule> rules, List<Atom> facts) {
+		super(rules, facts);
+		this.dependencyGraph = DependencyGraph.buildDependencyGraph(this.getRulesById());
+		this.componentGraph = this.buildComponentGraph(this.dependencyGraph);
+	}
+
+	public static AnalyzedProgram analyzeNormalProgram(NormalProgram prog) {
+		ImmutablePair<List<Atom>, List<InternalRule>> factsAndRules = InternalProgram.internalizeFactsAndRules(prog);
+		return new AnalyzedProgram(factsAndRules.right, factsAndRules.left);
+	}
+
+	private ComponentGraph buildComponentGraph(DependencyGraph depGraph) {
+		StronglyConnectedComponentsHelper sccHelper = new StronglyConnectedComponentsHelper();
+		SccResult sccResult = sccHelper.findStronglyConnectedComponents(depGraph);
```

Comment:
Theoretically that would be possible, however, since the SCC algorithm is quite complicated, it's easier to implement (and read) using an instance with some internal state (mainly storing results of DFS runs for later use in member variables rather than passing them around as method params). In order to keep `StronglyConnectedComponentsHelper` a bit simpler, I'd like to stay with the current version.